bots/succubus/succubus_main.lua

```diff
@@ -0,0 +1,584 @@
+		val = val + object.mesmeUpBonus
+	end
+	
+	if skills.hold:CanActivate() then
+		val = val + object.holdUpBonus
+	end
+
+	if skills.heartache:CanActivate() then
+		val = val + object.heartacheUpBonus
+	end
+
+	local unitSelf = core.unitSelf
+
+	if unitSelf:HasState("State_PowerupStealth") or unitSelf:HasState("State_PowerupMoveSpeed") then
+		val = val + 20
+	end
+
+	-- Less mana less aggerssion
+	val = val + (unitSelf:GetManaPercent() - 0.80) * 45
+	return val
+
+end
+behaviorLib.CustomHarassUtility = CustomHarassUtilityFnOverride  
+
+---------------------------------------------------------
+--					Harass Behavior					   --
+---------------------------------------------------------
+
+local function HarassHeroExecuteOverride(botBrain)
+	local unitSelf = core.unitSelf
+
+	--Cant trust to dontbreakchanneling
+	if unitSelf:IsChanneling() then
+		return true
+	end
+
+	local unitTarget = behaviorLib.heroTarget
+	if unitTarget == nil then
+		return object.harassExecuteOld(botBrain) --Target is invalid, move on to the next behavior
+	end
+	
+	--mesme goes where it wants
+	if unitTarget:HasState("State_Succubis_Ability3") then
+		return false
+	end
+
+	local vecMyPosition = unitSelf:GetPosition() 
+	local nAttackRange = core.GetAbsoluteAttackRangeToUnit(unitSelf, unitTarget)
+	local nMyExtraRange = core.GetExtraRange(unitSelf)
+	
+	local vecTargetPosition = unitTarget:GetPosition()
+	local nTargetDistanceSq = Vector3.Distance2DSq(vecMyPosition, vecTargetPosition)
+	local targetMagicImmune = unitTarget:isMagicImmune()
+	
+	local nLastHarassUtility = behaviorLib.lastHarassUtil
+	local bCanSee = core.CanSeeUnit(botBrain, unitTarget)	
+	local bActionTaken = false
+
+	local portalKey = core.GetItem("Item_PortalKey")
+	local puzzleBox = core.GetItem("Item_Summon")
+	local shrunkenHead = core.GetItem("Item_Immunity")
+
+	--damage stealth illusion movespeed regen
+	local runeInBottle = bottle.getRune()
+	if runeInBottle == "damage" or runeInBottle == "illusion" or runeInBottle == "movespeed" then
+		botBrain:OrderItem(core.GetItem("Item_Bottle"))
+	end
+
+	--pk suprise
+	if bCanSee and portalKey and portalKey:CanActivate() and object.pkThreshold < nLastHarassUtility then
+		if nTargetDistanceSq > 800 * 800 then
+			if nLastHarassUtility > behaviorLib.diveThreshold or core.NumberElements(core.GetTowersThreateningPosition(vecTargetPosition, nMyExtraRange, core.myTeam)) == 0 then
+				local _, sortedTable = HoN.GetUnitsInRadius(vecTargetPosition, 1000, core.UNIT_MASK_HERO + core.UNIT_MASK_ALIVE, true)
+				local EnemyHeroes = sortedTable.EnemyHeroes
+				if core.NumberElements(EnemyHeroes) == 1 then
+					bActionTaken = core.OrderItemPosition(botBrain, unitSelf, portalKey, vecTargetPosition)
+				end
+			end
+		end
+	end
+
+	local mesmeRange = skills.mesme:GetRange()
+	local smittenRange = skills.smitten:GetRange()
+	local mesmeCanActivate = skills.mesme:CanActivate()
+	local smittenCanActivate = skills.smitten:CanActivate()
+
+	--teamfight
```

Comment:
This code is extremely similar if not the same as above, so, move it into a function, with any differences as parameters